src/irr/asset/CImageLoaderOpenEXR.cpp

```diff
@@ -0,0 +1,454 @@
+
+			std::array<Array2D<half>, 4> halfPixelMapArray;
+			std::array<Array2D<float>, 4> fullFloatPixelMapArray;
+			std::array<Array2D<uint32_t>, 4> uint32_tPixelMapArray;
+
+			int width;
+			int height;
+			bool doesItSupportAlphaChannel;
+
+			ICPUImage::SCreationParams params;
+			params.type = ICPUImage::ET_2D;;
+			params.flags = static_cast<ICPUImage::E_CREATE_FLAGS>(0u);
+			params.samples = ICPUImage::ESCF_1_BIT;
+			params.extent.depth = 1u;
+			params.mipLevels = 1u;
+			params.arrayLayers = 1u;
+
+			specifyIrrlichtEndFormat(params.format, file, doesItSupportAlphaChannel);
+
+			if (params.format == EF_R16G16B16A16_SFLOAT || params.format == EF_R16G16B16_SFLOAT)
+				readRgba(file, halfPixelMapArray, width, height, params.format, doesItSupportAlphaChannel);
+			else if(params.format == EF_R32G32B32A32_SFLOAT || params.format == EF_R32G32B32_SFLOAT)
+				readRgba(file, fullFloatPixelMapArray, width, height, params.format, doesItSupportAlphaChannel);
+			else if (params.format == EF_R32G32B32A32_UINT || params.format == EF_R32G32B32_UINT)
+				readRgba(file, uint32_tPixelMapArray, width, height, params.format, doesItSupportAlphaChannel);
+
+			params.extent.width = width;
+			params.extent.height = height;
+
+			static const uint32_t MAX_PITCH_ALIGNMENT = 8u;											// OpenGL cannot transfer rows with arbitrary padding
+			auto calcPitchInBlocks = [](uint32_t width, uint32_t blockByteSize) -> uint32_t			// try with largest alignment first
+			{
+				auto rowByteSize = width * blockByteSize;
+				for (uint32_t _alignment = MAX_PITCH_ALIGNMENT; _alignment > 1u; _alignment >>= 1u)
+				{
+					auto paddedSize = core::alignUp(rowByteSize, _alignment);
+					if (paddedSize % blockByteSize)
+						continue;
+					return paddedSize / blockByteSize;
+				}
+				return width;
+			};
+
+			auto& image = ICPUImage::create(std::move(params));
+
+			auto texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(image->getImageDataSizeInBytes());
+			auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1u);
+			ICPUImage::SBufferCopy& region = regions->front();
+			//region.imageSubresource.aspectMask = ...; // waits for Vulkan
+			region.imageSubresource.mipLevel = 0u;
+			region.imageSubresource.baseArrayLayer = 0u;
+			region.imageSubresource.layerCount = 1u;
+			region.bufferOffset = 0u;
+			region.bufferRowLength = calcPitchInBlocks(width, getTexelOrBlockBytesize(image->getCreationParameters().format));
+			region.bufferImageHeight = 0u;
+			region.imageOffset = { 0u, 0u, 0u };
+			region.imageExtent = image->getCreationParameters().extent;
+
+			void* fetchedData = texelBuffer->getPointer();
+
+			for (uint64_t xPos = 0; xPos < width; ++xPos)
+				for (uint64_t yPos = 0; yPos < height; ++yPos)
+					for (uint8_t channelIndex = 0; channelIndex < (doesItSupportAlphaChannel ? 4 : 3); ++channelIndex)
+					{ 
+						const uint64_t ptrStyleEndShiftToImageDataPixel = (yPos * width) + xPos;
```

Comment:
This is what I refer to in https://github.com/buildaworldnet/IrrlichtBAW/pull/403/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=No+extension&hide-deleted-files=true#r358141441

Reply:
it was wronged, fixed without aligement